src/components/pages/IDPManagement/IDPList.tsx

```diff
@@ -425,15 +453,41 @@ export const IDPList = ({ isManagementOSP }: { isManagementOSP?: boolean }) => {
             isManagementOSP ? renderManagementOSPMenu(row) : renderMenu(row),
         },
       ]}
-      rows={
-        (isManagementOSP
-          ? idpsData?.filter(
-              (a) => a.identityProviderTypeId === IDPCategory.MANAGED
-            )
-          : idpsData) ?? []
-      }
+      rows={(isManagementOSP ? idpsManagedData : idpsData) ?? []}
       getRowId={(row: { [key: string]: string }) => row.identityProviderId}
       hasBorder={false}
+      searchPlaceholder={
+        isManagementOSP
+          ? t('content.onboardingServiceProvider.search')
+          : undefined
+      }
+      searchExpr={isManagementOSP ? searchExpr : undefined}
+      onSearch={
+        isManagementOSP
+          ? (expr: string) => {
+              isManagementOSP && onSearch(expr)
+              setSearchExpr(expr)
+            }
+          : undefined
+      }
+      searchDebounce={isManagementOSP ? 1000 : undefined}
+      onButtonClick={() => dispatch(show(OVERLAYS.ADD_IDP))}
+      buttonLabel={t('content.onboardingServiceProvider.addIdentityProvider')}
+      sx={
+        isManagementOSP
+          ? {
+              '.MuiDataGrid-columnHeadersInner': {
+                fontSize: '16px',
+                fontWeight: '400',
+                backgroundColor: '#E9E9E9',
+              },
+              '.MuiDataGrid-row': {
+                fontSize: '14px',
+                fontWeight: '400',
+              },
+            }
+          : undefined
+      }
```

Comment:
This looks ugly - can you please define that styling outside in an SxProps var or use a function for the entire ternary expression